homeassistant/components/ohme/coordinator.py

```diff
@@ -0,0 +1,71 @@
+"""Ohme coordinators."""
+
+from datetime import timedelta
+import logging
+
+from ohme import ApiException, AuthException, OhmeApiClient
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError, ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+type OhmeConfigEntry = ConfigEntry[OhmeCoordinator]
+
+
+class OhmeCoordinator(DataUpdateCoordinator[None]):
+    """Coordinator to pull all updates from the API."""
+
+    config_entry: OhmeConfigEntry
+
+    def __init__(self, hass: HomeAssistant, entry: OhmeConfigEntry) -> None:
+        """Initialise coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="Ohme Coordinator",
+            update_interval=timedelta(seconds=30),
+            config_entry=entry,
+        )
+
+        self.client: OhmeApiClient = OhmeApiClient(
+            self.config_entry.data[CONF_EMAIL], self.config_entry.data[CONF_PASSWORD]
+        )
+        self._alternative_iteration: bool = True
+
+    async def _async_setup(self) -> None:
+        try:
+            await self.client.async_login()
+
+            if not await self.client.async_update_device_info():
+                raise ConfigEntryNotReady(
+                    translation_key="device_info_failed", translation_domain=DOMAIN
+                )
+        except AuthException as e:
+            raise ConfigEntryError(
+                translation_key="auth_failed", translation_domain=DOMAIN
+            ) from e
+        except ApiException as e:
+            raise ConfigEntryNotReady(
+                translation_key="api_failed", translation_domain=DOMAIN
+            ) from e
+
+    async def _async_update_data(self) -> None:
+        """Fetch data from API endpoint."""
+        try:
+            await self.client.async_get_charge_session()
+
+            # Fetch on every other update
+            if self._alternative_iteration:
+                await self.client.async_get_advanced_settings()
```

Comment:
What's this for?

Reply:
Some states (currently only CT reading) come from a different API endpoint. This is just to fetch that little less often, and save using a second coordinator

Reply:
Your original code used to have 4, how will you add the other 2?

Reply:
3 of those 4 need to be updated frequently (30s/1m), with the fourth being less frequent (10m). 

Aiming to keep things as simple as possible, my proposed solution in future PRs would be to make the 3 requests in OhmeCoordinator then add a second coordinator for the last.

Reply:
I think i secretly like a separate coördinator for it. We for example have a service action to update the state of an entity, and that service now can't properly be used on one of the 1 minute entities as it will only work every other service action execution. And the other way around, if someone uses that service on the 30s one to request faster or slower, you automatically make the other request faster or slower as well. 

So like, yes I do see why you did this, but I think with this extra info I'm not sure this would make a good experience

Reply:
So both coordinators are going to be calling methods of OhmeApiClient, and that is setup in OhmeCoordinator._async_setup. Whats the best way to have a second coordinator that depends on the actions of the first?

Reply:
You would then execute those actions in `__init__.py` and then inject the client into the coordinator. (and then you won't implement `_async_setup`